python/databases/inversekinematics.py

```diff
@@ -295,15 +295,33 @@ def clone(self,envother):
     def has(self):
         return self.iksolver is not None and self.manip.GetIkSolver() is not None and self.manip.GetIkSolver().Supports(self.iktype) and self.iksolver.GetXMLId() == self.manip.GetIkSolver().GetXMLId()
     
-    def save(self):
+    def save(self, filepermissions = None):
         statsfilename=self.getstatsfilename(False)
         try:
-            os.makedirs(os.path.split(statsfilename)[0])
-        except OSError:
-            pass
-        with open(statsfilename, 'w') as f:
-            pickle.dump((self.getversion(),self.statistics,self.ikfeasibility,self.solveindices,self.freeindices,self.freeinc), f)
-        log.info('inversekinematics generation is done, compiled shared object: %s',self.getfilename(False))
+            defaultMask = os.umask(0)
+            basepath = os.path.split(statsfilename)[0]
+            try:
+                os.makedirs(basepath)
+            except OSError as e:
+                pass
+
+            with open(statsfilename, 'w') as f:
+                pickle.dump((self.getversion(),self.statistics,self.ikfeasibility,self.solveindices,self.freeindices,self.freeinc), f)
+            log.info('inversekinematics generation is done, compiled shared object: %s',self.getfilename(False))
+
+            if filepermissions is not None and filepermissions >= 0:
+                log.info('changing filepermissions of path %s to \'%s\' recursively.', basepath, filepermissions)
+                try:
+                    os.chmod(basepath, filepermissions)
+                    for root, dirs, files in os.walk(basepath):
+                        for d in dirs:
+                            os.chmod(os.path.join(root, d), filepermissions)
+                        for f in files:
+                            os.chmod(os.path.join(root, f), filepermissions)
+                except OSError:
+                    pass
```

Comment:
@AntoineRyu can you print the error so that we know when permissions fail? thanks!

Reply:
Done